tests/functional/test_import.py

```diff
@@ -47,14 +47,105 @@ def test_ctypes_CDLL_c(pyi_builder):
         """)
 
 
-# libnss_files.so should be available in most Unix/Linux systems using
-# glibc
-@skipif(not ctypes.util.find_library('nss_files'),
-        reason="required libnss_files.so missing")
-def test_ctypes_CDLL_find_library__nss_files(pyi_builder):
-    pyi_builder.test_source(
-        """
-        import ctypes, ctypes.util
-        lib = ctypes.CDLL(ctypes.util.find_library('nss_files'))
-        assert lib is not None
-        """)
+def skip_if_lib_missing(libname, text=None):
+    """
+    pytest decorator to evaluate the required shared lib.
+
+    :param libname: Name of the required library.
+    :param text: Text to put into the reason message
+                 (defaults to 'lib%s.so' % libname)
+
+    :return: pytest decorator with a reason.
+    """
+    soname = ctypes.util.find_library(libname)
+    if not text:
+        text = "lib%s.so" % libname
+    # Return pytest decorator.
+    return skipif(not (soname and ctypes.CDLL(soname)),
+                  reason="required %s missing" % text)
+
+
+_template_ctypes_CDLL_find_library = """
+    import ctypes, ctypes.util, sys, os
+    lib = ctypes.CDLL(ctypes.util.find_library(%(libname)r))
+    print(lib)
+    assert lib is not None and lib._name is not None
+    if getattr(sys, 'frozen', False):
+        soname = ctypes.util.find_library(%(libname)r)
+        print(soname)
+        libfile = os.path.join(sys._MEIPASS, soname)
+        print(libfile)
+        assert os.path.isfile(libfile), '%%s is missing' %% soname
+        print('>>> file found')
+    """
+
+_template_ctypes_test = """
+        print(lib)
+        assert lib is not None and lib._name is not None
+        import sys, os
+        if getattr(sys, 'frozen', False):
+            libfile = os.path.join(sys._MEIPASS, %(soname)r)
+            print(libfile)
+            assert os.path.isfile(libfile), '%(soname)s is missing'
+            print('>>> file found')
+    """
+
+# Ghostscript's libgs.so should be available in may Unix/Linux systems
+@skip_if_lib_missing('gs', 'libgs.so (Ghostscript)')
+def test_ctypes_CDLL_find_library__gs(pyi_builder):
+    libname = 'gs'
+    pyi_builder.test_source(_template_ctypes_CDLL_find_library % locals())
+
+
+#-- Generate test-cases for the different types of ctypes objects.
+
+libname = 'gs'
+reason = 'libgs.so (Ghostscript)'
+parameters = []
+ids = []
+for prefix in ('', 'ctypes.'):
+    for funcname in  ('CDLL', 'PyDLL', 'WinDLL', 'OleDLL', 'cdll.LoadLibrary'):
+        ids.append('%s_%s' % (prefix+funcname, libname))
+        params = (prefix+funcname, libname, reason, ids[-1])
+        if funcname in ("WinDLL", "OleDLL"):
+            # WinDLL, OleDLL only work on windows.
+            params = skipif_notwin(params)
```

Comment:
I tested this, but I can not verify your statement. Here is the code I tried:
```python
import pytest
import os

T1 = bool(os.getenv('T1'))
T2 = bool(os.getenv('T2'))
T3 = bool(os.getenv('T3'))

parameters = []
for i, prefix in enumerate(('blabla', 'lalala', 'abcdef')):
    params = (i, prefix)
    if i == 1:
        params = pytest.mark.skipif(not T3, reason="requires T3")(params)
    parameters.append(params)

@pytest.mark.skipif(not T2, reason="requires T2")
@pytest.mark.parametrize("num, prefix", parameters)
@pytest.mark.skipif(not T1, reason="requires T1")
def test_answer(num, prefix):
    pass
```
and running it like:
```
$ T1=yes T2=yes T3= py.test -rs test_sample2.py 
================================= test session starts =================================
platform linux -- Python 3.3.2 -- py-1.4.30 -- pytest-2.7.2
rootdir: /tmp, inifile: 
collected 3 items 

test_sample2.py .s.
=============================== short test summary info ===============================
SKIP [1] test_sample2.py:15: requires T3

========================= 2 passed, 1 skipped in 0.01 seconds =========================
```